src/network/gossip.rs

```diff
@@ -248,18 +303,46 @@ impl SnapchainGossip {
         }
     }
 
+    pub fn publish_contact_info(&mut self) {
+        let contact_info = ContactInfo {
+            body: Some(ContactInfoBody {
+                peer_id: self.swarm.local_peer_id().to_bytes(),
+                gossip_address: self.announce_address.clone(),
+                network: self.fc_network as i32,
+                snapchain_version: PROTOCOL_VERSION.to_string(),
+                timestamp: std::time::SystemTime::now()
+                    .duration_since(std::time::UNIX_EPOCH)
+                    .unwrap()
+                    .as_millis() as u64,
+            }),
+        };
+
+        let gossip_message = GossipMessage {
+            gossip_message: Some(gossip_message::GossipMessage::ContactInfoMessage(
+                contact_info,
+            )),
+        };
+        self.publish(gossip_message.encode_to_vec(), CONTACT_INFO);
```

Comment:
I think validators shouldn't publish contactInfo, we don't want to induce extra connection than what happens at bootstrap